### PR TITLE
Misc Compiler Optimizations

### PR DESCRIPTION
**Pull request by ike709**
These tiny changes shave about ~1050 ms off of TG compilation, bringing it from ~10.8 seconds to ~9.7 seconds on my computer.

### CODE REVIEW

File: DMCompiler/Compiler/DMPreprocessor/DMPreprocessor.cs [hunk 1]
```diff
@@ -15,18 +15,18 @@ namespace DMCompiler.Compiler.DMPreprocessor {
...
 
-        private HashSet<string> _includedFiles = new();
-        private Stack<Token> _unprocessedTokens = new();
+        private readonly HashSet<string> _includedFiles = new(5120); // Capacity Note: TG peaks at 4860 at time of writing
+        private readonly Stack<Token> _unprocessedTokens = new(8192); // Capacity Note: TG peaks at 6802 at time of writing
         private bool _currentLineWhitespaceOnly = true;
-        private bool _enableDirectives;
-        private Dictionary<string, DMMacro> _defines = new() {
+        private readonly bool _enableDirectives;
+        private readonly Dictionary<string, DMMacro> _defines = new(12288) { // Capacity Note: TG peaks at 9827 at time of writing. Current value is arbitrarily 4096 * 3.
```
Comment (ike709) [inline DMCompiler/Compiler/DMPreprocessor/DMPreprocessor.cs:29]: Setting these initial capacities accounts for about ~900 ms of total savings.

File: DMCompiler/Compiler/DM/DMLexer.cs [hunk 2]
```diff
@@ -5,8 +5,9 @@
 using OpenDreamShared.Compiler;
 
 namespace DMCompiler.Compiler.DM {
-    public partial class DMLexer : TokenLexer {
-        public static List<string> ValidEscapeSequences = new() {
+    public sealed class DMLexer : TokenLexer {
+        // NOTE: .NET still needs you to pass the capacity size to generate the most optimal code, so update it when you change these values
+        public static readonly List<string> ValidEscapeSequences = new(41) {
```
Comment (wixoaGit) [inline DMCompiler/Compiler/DM/DMLexer.cs:10]: Is it really much of a concern when the variable is static? It's only every going to be created once, when the application starts.

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: DMCompiler/Compiler/DM/DMLexer.cs (hunk 2 above)
Comment (ike709) [reply DMCompiler/Compiler/DM/DMLexer.cs:10]: Is there any reason _not_ to? How often does this list really get edited and how much work is it to just change the number?

Comment (wixoaGit) [reply DMCompiler/Compiler/DM/DMLexer.cs:10]: Fair enough

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**